synapse/tests/utils.py

```diff
@@ -449,7 +473,49 @@ def write(self, s):
         if self.mesg and self.mesg in s:
             self.set()
 
+class AsyncStreamEvent(io.StringIO, asyncio.Event):
+    '''
+    A combination of a io.StringIO object and a threading.Event object.
+    '''
+    def __init__(self, *args, **kwargs):
+        io.StringIO.__init__(self, *args, **kwargs)
+        asyncio.Event.__init__(self, loop=asyncio.get_running_loop())
+        self.mesg = ''
+
+    def setMesg(self, mesg):
+        '''
+        Clear the internal event and set a new message that is used to set the event.
+
+        Args:
+            mesg (str): The string to monitor for.
+
+        Returns:
+            None
+        '''
+        self.mesg = mesg
+        self.clear()
+
+    def write(self, s):
+        io.StringIO.write(self, s)
+        if self.mesg and self.mesg in s:
+            self.set()
+
+    async def wait(self, timeout=None):
+        if timeout is None:
+            return await asyncio.Event.wait(self)
+        return await s_coro.event_wait(self, timeout=timeout)
+
 class SynTest(unittest.TestCase):
+    '''
+    Mark all async test methods as s_glob.synchelp decorated
+    '''
+    def __init__(self, *args, **kwargs):
+        unittest.TestCase.__init__(self, *args, **kwargs)
+        for s in dir(self):
+            attr = getattr(self, s, None)
+            # If s is an instance method and starts with 'test_', synchelp wrap it
```

Comment:
I need to do some more testng with this but I do believe it removes our ability to run a single target test with unittest.

Reply:
confirmed - I cannot run a specific unit test.  I can run a unit test if i remove this comment and add the explicit synchelp decorator.
```
synapse$ python -m unittest synapse.tests.test_cortex.CortexTest.test_pivot_inout
/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)
Traceback (most recent call last):
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/unittest/__main__.py", line 18, in <module>
    main(module=None)
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/unittest/main.py", line 147, in parseArgs
    self.createTests()
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/unittest/main.py", line 159, in createTests
    self.module)
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/unittest/loader.py", line 220, in loadTestsFromNames
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/unittest/loader.py", line 220, in <listcomp>
    suites = [self.loadTestsFromName(name, module) for name in names]
  File "/home/lol/.pyenv/versions/3.7-dev-debug/lib/python3.7/unittest/loader.py", line 205, in loadTestsFromName
    test = obj()
TypeError: test_pivot_inout() missing 1 required positional argument: 'self'
```

Reply:
Pytest does work but it is not a explicit synapse requirement
`pytest synapse/tests/test_cortex.py::CortexTest::test_pivot_inout`